### ISSUE TITLE
dyld troubles on macOS Mojave with mysql2 0.4.x

### ISSUE THREAD

**Issue by thegcat** [brianmario/mysql2#1001]
When starting an existing rails 4 app on macOS Mojave with mysql2 0.4.6 (the results with mysql2 0.4.10 are the same):

```
dyld: lazy symbol binding failed: Symbol not found: _mysql_server_init
  Referenced from: /Users/thegcat/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/mysql2-0.4.6/lib/mysql2/mysql2.bundle
  Expected in: flat namespace

dyld: Symbol not found: _mysql_server_init
  Referenced from: /Users/thegcat/.rbenv/versions/2.3.7/lib/ruby/gems/2.3.0/gems/mysql2-0.4.6/lib/mysql2/mysql2.bundle
  Expected in: flat namespace

fish: 'rails s' terminated by signal SIGABRT (Abort)
```

I'm guessing the issue is related to some paths not being set correctly, but honestly I'm a bit at a loss here.

The mysql installation is a mysql 5.6 installed via homebrew, though the same errors also arise with mysql 5.7.

Different things I have tried, including building with `--with-mysql-rpath=/usr/local/opt/mysql@5.6/lib`, `brew link --force`-ing, exporting the mysql lib path in `DYLD_LIBRARY_PATH` amongst others did not yield a different result than above.

For the record: I had also tried mysql2 0.5.2, but as this won't even load into rails 4 I can't say wether it works better or not.

Any pointers on how to tackle this issue?

**Comment by sodabrew** [comment]
Could you try the latest point release of 0.4.x and see if that helps?

**Comment by thegcat** [comment]
I tried the different steps with 0.4.10 too and got the same error as written above. I have just tried it again to confirm, the error is the same.

**Comment by thegcat** [comment]
During the installation of the gem following message happens:

```
Don't know how to set rpath on your system, if MySQL libraries are not in path mysql2 may not load
```

Maybe this helps?

**Comment by thegcat** [comment]
I have in the meantime completely nuked my home-brew install as well as my rbenv install and installed both and everything in that anew on macOS 10.14, sadly to no avail neither with mysql2 0.4.6 nor 0.4.10.

**Comment by tesssie** [comment]
@thegcat  I had this issue.  Hope you have xcode command line installed. I had to to install some missing header files by 
`open /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg`

This linked helped me [](https://stackoverflow.com/questions/52509602/cant-compile-c-program-on-a-mac-after-upgrade-to-mojave/52530212#)52530212

**Comment by thegcat** [comment]
@tesssie thanks a lot for the suggestion, after recompiling the mysql2 gems this seems to work.

**Comment by tesssie** [comment]
@thegcat  recompiling manually?

**Comment by thegcat** [comment]
Not really, just `gem uninstall mysql2` and `bundle install` to install it back.

**Comment by tesssie** [comment]
oh  :)

**Comment by D-Maher** [comment]
@tesssie, thanks for your solution! This worked like a charm for me.

**Comment by patriciojofre** [comment]
Thank you @tesssie , you save my life!!

**Comment by dkassen** [comment]
@tesssie you're a geeeeenius

**Comment by gilangmugnirespaty** [comment]
Thank you @tesssie
